src/jit/lower.cpp

```diff
@@ -3123,6 +3133,195 @@ void Lowering::LowerAdd(GenTreePtr* pTree, Compiler::fgWalkData* data)
 #endif // !_TARGET_ARMARCH_
 }
 
+//------------------------------------------------------------------------
+// LowerUDivUMod: transform GT_UDIV/GT_UMOD nodes with a const power of 2
+// divisor into GT_RSZ/GT_AND nodes.
+//
+// Arguments:
+//    tree:   pointer to the GT_UDIV/GT_UMOD node to be lowered
+
+void Lowering::LowerUDivUMod(GenTree* tree)
+{
+    GenTreeOp* divmod = tree->AsOp();
```

Comment:
Please assert that tree is a udiv or umod

assert(tree->OperGet() == GT_UDIV || tree->OperGet() == GT_UMOD);